src/prompt.c

```diff
@@ -88,3 +93,38 @@ void	prompt_destroy(t_prompt *_prompt)
 		free(_prompt->buffer);
 	free(_prompt);
 }
+
+void	prompt_update(t_prompt *_prompt)
+{
+	prompt_update_current_path(_prompt);
+	prompt_prepare_buffer(_prompt);
+}
+
+char	*prompt_read(void)
+{
+	char	*command_line;
+
+	if (!(get_next_line(STDIN, &command_line) > 0))
```

Comment:
are you sure get_next_line doesn't return malloced memory when it returns 0 or below 0 ?